src/statistician.jl

```diff
@@ -12,90 +14,87 @@ There are six parts:
 - latent instance encoder q(z|x,c)
 - decoder p(x|z)
 """
-
-struct Statistician{IE,pc <: ContinuousMultivariateDistribution,qc <: ConditionalMvNormal,pz <: ConditionalMvNormal,qz <: ConditionalMvNormal,D <: ConditionalMvNormal} # <: AbstractNS
-	instance_encoder::IE
-	prior_c::pc
-	encoder_c::qc
-	conditional_z::pz
-	encoder_z::qz
-	decoder::D
+struct NeuralStatistician{IE,pc <: ContinuousMultivariateDistribution,qc <: ConditionalMvNormal,pz <: ConditionalMvNormal,qz <: ConditionalMvNormal,D <: ConditionalMvNormal} # <: AbstractNS
+    instance_encoder::IE
+    prior_c::pc
+    encoder_c::qc
+    conditional_z::pz
+    encoder_z::qz
+    decoder::D
 end
 
-Flux.@functor Statistician
+Flux.@functor NeuralStatistician
 
-function Flux.trainable(m::Statistician)
-	(instance_encoder = m.instance_encoder, encoder_c = m.encoder_c, conditional_z = m.conditional_z, encoder_z = m.encoder_z, decoder = m.decoder)
+function Flux.trainable(m::NeuralStatistician)
+    (instance_encoder = m.instance_encoder, encoder_c = m.encoder_c, conditional_z = m.conditional_z, encoder_z = m.encoder_z, decoder = m.decoder)
 end
 
-function Statistician(clength::Int, IE, enc_c::ConditionalMvNormal, cond_z::ConditionalMvNormal, enc_z::ConditionalMvNormal, dec::ConditionalMvNormal)
-	W = first(Flux.params(enc_c))
+function NeuralStatistician(clength::Int, IE, enc_c::ConditionalMvNormal, cond_z::ConditionalMvNormal, enc_z::ConditionalMvNormal, dec::ConditionalMvNormal)
+    W = first(Flux.params(enc_c))
     μ_c = fill!(similar(W, clength), 0)
     σ_c = fill!(similar(W, clength), 1)
     prior_c = DistributionsAD.TuringMvNormal(μ_c, σ_c)
-	Statistician(IE, prior_c, enc_c, cond_z, enc_z, dec)
+    NeuralStatistician(IE, prior_c, enc_c, cond_z, enc_z, dec)
 end
 
 
 """
-    lossNS(x::AbstractArray,model::Statistician; β1=1.0, β2=1.0)
+    elbo(m::NeuralStatistician,x::AbstractArray; β1=1.0, β2=1.0)
 
-Neural Statistician loss.
+Neural Statistician log-likelihood lower bound.
 
-For NS a Statistician model, simply create a loss
+For a Neural Statistician model, simply create a loss
 function as
     
-    `loss(x) = lossNS(x, NS)`
+    `loss(x) = -elbo(NeuralStatistician,x)`
```

Comment:
This could be misunderstood. It looks like the `elbo` would only get a `NeuralStatistician` type instead of an actual object. maybe something like this:
```julia
model = NeuralStatistician(cdim, instance_enc, enc_c_dist, cond_z_dist, enc_z_dist, dec_dist)
loss(x) = - elbo(model,x)
```
In that case it would also be really great to have short explanations of the variables in there like `instance_enc`. but if you are pressed with time we can just merge it with the corrections above.